packages/block-library/src/columns/edit.js

```diff
@@ -102,17 +102,29 @@ function ColumnsEditContainer( {
 			</InspectorControls>
 			{ InspectorControlsColorPanel }
 			<BackgroundColor>
-				<TextColor>
-					<InnerBlocks
-						allowedBlocks={ ALLOWED_BLOCKS }
-						__experimentalMoverDirection="horizontal"
-						ref={ ref }
-						__experimentalTagName={ Block.div }
-						__experimentalPassedProps={ {
-							className: classes,
-						} }
-					/>
-				</TextColor>
+				{ ( backgroundProps ) => (
+					<TextColor>
+						{ ( textColorProps ) => (
+							<InnerBlocks
+								allowedBlocks={ ALLOWED_BLOCKS }
+								__experimentalMoverDirection="horizontal"
+								ref={ ref }
+								__experimentalTagName={ Block.div }
+								__experimentalPassedProps={ {
```

Comment:
I've been wondering how to best improve this API... Maybe we should just spread the props like we do for `RichText`.

Reply:
@aduth do you have thoughts here?

Reply:
Could we just have a render prop?

```jsx
<InnerBlocks
    allowedBlocks={ ALLOWED_BLOCKS }
    __experimentalMoverDirection="horizontal"
    ref={ ref }
>
    { ( { children } ) => (
        <div 
            className={ classnames(
                classes,
                backgroundProps.className,
                textColorProps.className
            ) }
            style={ {
                ...backgroundProps.style,
                ...textColorProps.style,
            } }
        >
            { children }
        </div>
    ) }
</InnerBlocks>
```

Reply:
The columns block doesn't. have any other wrapper rendered by InnerBlocks, so that div rendered inside the render function here should be the same one rendered by the BlockList component. I believe a render prop makes this more complex.